client/fingerprint/env_aws.go

```diff
@@ -77,12 +77,16 @@ func (f *EnvAWSFingerprint) Fingerprint(request *FingerprintRequest, response *F
 		timeout = 1 * time.Millisecond
 	}
 
-	ec2meta, err := ec2MetaClient(f.endpoint, timeout)
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+
+	imdsClient, err := imdsClient(ctx, f.endpoint)
```

Comment:
Where does `f.endpoint` actually get set if we're no longer configuring it in the constructor? It looks like there's possibly fallback behavior if it's empty (maybe the client library reads the env var if set?), but are we ever using the `endpoint` field if we're falling back to that?

Reply:
I was actually meaning to leave a comment around this.  I took it out here because from what I can tell, AWS makes no mention of `AWS_ENV_URL`, and I don't think you can specify custom IMDS endpoints.  So that really just leaves overriding the endpoint for testing.

Reply:
+1 on some comments here; took a moment of digging around to see what was happening, otherwise it looks a bit suspect on the surface.